coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -482,41 +537,138 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
+              //
+              // Given we have offset of 10 mins, the query range from partition P1 (left of the partition split point)
+              // is [s, p + 10m]. The offset looks at data 10 mins back, so we can extent the time range in p1 to 10
+              // mins after the split point p We want to now provide results for time range t1 - t2, which is missing
+              // Lets assume the query is sum(rate(foo{}[5m] offset 10m))
+              // Given the offset is 10m, lookback is 5m, we would need raw data in the range
+              // [t1 - 5m - 10m, t2], this range for raw queries span two partitions and we will let the RawSeries (the
+              // leaf logical plan) with supportsRemoteDataCall = true figure out if this range can entirely be selected
+              // from partition p1 or p2
+              //
+
+              // Do not perform raw exports if the export is beyond a certain value for example
+              // foo{}[10d] or foo[2d] offset 8d  both will export 10 days of raw data which might cause heap pressure
+              // and OOMs. The max cross partition raw export config can control such queries from bring the process
+              // down but simpler queries with few minutes or even hour or two of lookback/offset will continue to work
+              // seamlessly with no data gaps
+              // Note that at the moment, while planning, we only can look at whats the max time range we can support.
+              // We still dont hqve capabilities to check the expected number of timeseries scanned or bytes scanned
+              // and adding capabilities to give up a "part" of query execution if the runtime number of bytes of ts
+              // scanned goes high isn't available. To start with the time range scanned as a static configuration will
+              // be good enough and can be enhanced in future as required.
+              val totalExpectedRawExport = (endTime - startTime) + lookbackMs + offsetMs
+              if (queryConfig.maxRemoteRawExportTimeRange.toMillis > totalExpectedRawExport) {
+                // Only if the raw export is completely within the previous partition's timerange
+                val newParams = qParams.copy(startSecs = startTime / 1000, endSecs = endTime / 1000)
+                val newContext = qContext.copy(origQueryParams = newParams)
+                val newLp = rewritePlanWithRemoteRawExport(logicalPlan, IntervalSelector(startTime, endTime))
+                ep ++= walkLogicalPlanTree(newLp, newContext, true).plans
+              } else {
+                logger.warn(
+                  s"Remote raw export is supported but the expected raw export for $totalExpectedRawExport ms" +
+                  s" is greater than the max allowed raw export duration ${queryConfig.maxRemoteRawExportTimeRange}")
+              }
+            }
+            val newParams = qParams.copy(startSecs = currentTimeRange.startMs / 1000,
```

Comment:
I can add but since the ``currentTimeRange`` is one of the assignmentRanges which already takes care to stamp the appropriate start and end time, I left it as is.  Let me add the max too